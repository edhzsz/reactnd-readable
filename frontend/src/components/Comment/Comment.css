.comment {
  overflow: hidden;
}

.comment-content {
  float:left;
  margin: 10px;
}

.comment-content span {
  display: block;
}
