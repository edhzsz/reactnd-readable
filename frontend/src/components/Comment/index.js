import React from 'react';
import PropTypes from 'prop-types';
import './Comment.css'
import Votes from '../Votes'

const Comment = (props) => {
  return (
    <div className="comment">
      <Votes votes={props.votes} onUpvote={props.onUpvote} onDownvote={props.onDownvote} size={20} />
      <div className="comment-wrapper">
        <span className="comment-content">{props.content}</span>
        <span className="comment-author">{props.author}</span>
      </div>
    </div>
  );
}

Comment.propTypes = {
  author: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  votes: PropTypes.number.isRequired,
  onUpvote: PropTypes.func,
  onDownvote: PropTypes.func
}

export default Comment;