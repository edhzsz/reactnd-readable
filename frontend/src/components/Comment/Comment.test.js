import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Comment from '.';

const testComment = {
  id: '894tuq4ut84ut8v4t8wun89g',
  parentId: "8xf0y6ziyjabvozdd253nd",
  timestamp: 1468166872634,
  body: 'Hi there! I am a COMMENT.',
  author: 'thingtwo',
  voteScore: 6,
  deleted: false,
  parentDeleted: false
};

const wrapper = shallow(<Comment comment={testComment}/>);

describe('Component: Comment', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Comment comment={testComment}/>, div);
  });

  it('should match its snapshot with required props', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});