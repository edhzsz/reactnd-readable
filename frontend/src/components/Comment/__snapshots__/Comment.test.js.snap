// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Comment should match its snapshot with required props 1`] = `
<div
  className="comment"
>
  <Votes
    size={20}
    votes={6}
  />
  <div
    className="comment-content"
  >
    <span
      className="comment-body"
    >
      Hi there! I am a COMMENT.
    </span>
    <span
      className="comment-author"
    >
      thingtwo
    </span>
    <Timestamp
      className="comment-timestamp"
      component="span"
      format="full"
      includeDay={false}
      precision={1}
      since={null}
      style={Object {}}
      time={1468166872.634}
      until={null}
      utc={true}
    />
  </div>
</div>
`;
